pkg/scheduler/service/service_test.go

```diff
@@ -90,3 +90,25 @@ func Test_awaitFailoverDoneFailover(t *testing.T) {
 		t.Fatalf("expected call to failover handler")
 	}
 }
+
+func Test_validateLeadershipTransition(t *testing.T) {
+	if err := validateLeadershipTransition("1_name-1", ""); err != nil {
+		t.Errorf("expected validation to pass when current id is empty")
+	}
+
+	if err := validateLeadershipTransition("1_name-1", "1_name-2"); err != nil {
+		t.Errorf("expected validation to pass when desired group equals current group")
+	}
+
+	if err := validateLeadershipTransition("0_name-1", ""); err == nil {
```

Comment:
see comments in validation func re: 0 checks - they were originally intended as "unparseable" checks